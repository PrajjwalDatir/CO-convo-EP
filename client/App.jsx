import React, { Suspense } from "react";
import { hot } from "react-hot-loader/root";
import { Switch, Route } from "react-router-dom";

const rootStyles = { fontFamily: "Lato, sans-serif" };

// routes are lazy loaded to split up bundle and reduce initial
// loading time.
// https://reactjs.org/docs/code-splitting.html
function App() {
  return (
    <div className="text-gray-900" style={rootStyles}>
      <Suspense fallback={null}>
        <Switch>
          <Route
            exact
            path="/help"
            component={React.lazy(() =>
              import(/* webpackChunkName: "chat" */ "./routes/help/help")
            )}
          />
          <Route
            exact
            path="/legal"
            component={React.lazy(() =>
              import(/* webpackChunkName: "chat" */ "./routes/legal/terms")
            )}
          />
          <Route
            exact
            path="/settings"
            component={React.lazy(() =>
              import(
                /* webpackChunkName: "chat" */ "./routes/chat/appBar/settings"
              )
            )}
          />
          <Route
            exact
            path="/chat"
            component={React.lazy(() =>
              import(/* webpackChunkName: "chat" */ "./routes/chat")
            )}
          />
          <Route
            exact
            path="/register"
            component={React.lazy(() =>
              import(/* webpackChunkName: "register" */ "./routes/register")
            )}
          />
          <Route
            exact
            path="/login"
            component={React.lazy(() =>
              import(/* webpackChunkName: "login" */ "./routes/login")
            )}
          />
          <Route
            exact
            path="/"
            component={React.lazy(() =>
              import(/* webpackChunkName: "home" */ "./routes/home")
            )}
          />
          <Route
            path="*"
            component={React.lazy(() =>
              import(/* webpackChunkName: "notfound" */ "./routes/notfound")
            )}
          />
        </Switch>
      </Suspense>
    </div>
  );
}

export default hot(App);
