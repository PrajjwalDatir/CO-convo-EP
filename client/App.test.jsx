import React from 'react';
import { mount } from 'enzyme';
import { StaticRouter } from 'react-router-dom';
import App from './App';

describe('<App />', () => {
  it('should match snapshot', () => {
    const wrapper = mount(
      <StaticRouter>
        <App />
      </StaticRouter>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
