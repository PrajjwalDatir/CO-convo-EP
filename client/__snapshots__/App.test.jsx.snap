// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<StaticRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    staticContext={Object {}}
  >
    <App>
      <div
        className="text-gray-900"
        style={
          Object {
            "fontFamily": "Lato, sans-serif",
          }
        }
      >
        <Suspense
          fallback={null}
        >
          <Switch>
            <Route
              component={
                Object {
                  "$$typeof": Symbol(react.lazy),
                  "_ctor": [Function],
                  "_result": Promise {},
                  "_status": 0,
                }
              }
              computedMatch={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
              exact={true}
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              path="/"
            />
          </Switch>
        </Suspense>
      </div>
    </App>
  </Router>
</StaticRouter>
`;
